meilisearch/tests/index/test_index.py

```diff
@@ -73,60 +93,71 @@ def test_get_or_create_index(self):
         assert len(documents) == 1
         index_2.delete()
         with pytest.raises(Exception):
-            self.client.get_index(index_3).info()
+            self.client.get_index(index_3)
 
     def test_get_or_create_index_with_primary_key(self):
         """Test get_or_create_index method with primary key"""
         index_1 = self.client.get_or_create_index('books', {'primaryKey': self.index_uid4})
         index_2 = self.client.get_or_create_index('books', {'primaryKey': 'some_wrong_key'})
+        assert index_1.primary_key == self.index_uid4
         assert index_1.get_primary_key() == self.index_uid4
+        assert index_2.primary_key is None
         assert index_2.get_primary_key() == self.index_uid4
+        assert index_2.primary_key == self.index_uid4
         index_1.delete()
 
-    def test_index_info(self):
-        """Tests getting an index's info"""
-        index = self.client.get_index(uid=self.index_uid)
-        response = index.info()
+    def test_index_fetch_info(self):
+        """Tests getting the index info"""
+        index = self.client.index(uid=self.index_uid)
+        response = index.fetch_info()
         assert isinstance(response, object)
-        assert response['uid'] == self.index_uid
-        assert response['primaryKey'] is None
-
-    def test_index_info_with_wrong_uid(self):
-        """Tests getting an index's info in MeiliSearch with a wrong UID"""
-        with pytest.raises(Exception):
-            self.client.get_index(uid='wrongUID').info()
+        assert response.uid == self.index_uid
+        assert response.primary_key is None
+        assert response.primary_key == index.primary_key
+        assert response.primary_key == index.get_primary_key()
+
+    def test_index_fetch_info_containing_primary_key(self):
+        """Tests getting the index info"""
+        index = self.client.index(uid=self.index_uid3)
+        response = index.fetch_info()
+        assert isinstance(response, object)
```

Comment:
```suggestion
        assert isinstance(response, Index)
```
I still wonder what is the exact purpose of this assert, checking that it is an instance of `object`. In any case, the `index(...)` method returns an Index instance or raises an Error, so maybe this assert is not necessary. If we leave it, we can check that it returns an Index instance

Reply:
I would do it, but impossible, the `meilisearch.Index` module is not accessible during the import.